src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -3339,8 +3315,11 @@ private IEnumerable<SourceRepository> GetEffectiveSources(IEnumerable<SourceRepo
         {
             // Always have to add the packages folder as the primary repository so that
             // dependency info for an installed package that is unlisted from the server is still available :(
-            var effectiveSources = new List<SourceRepository>(primarySources);
-            effectiveSources.Add(PackagesFolderSourceRepository);
+            var effectiveSources = new List<SourceRepository>(primarySources)
```

Comment:
it would be better to add all of these in one go. or atleast allocate the list size accordingly